src/porepy/numerics/fracture_deformation.py

```diff
@@ -1,365 +1,11 @@
-        in direction of shear traction as:
-        d += T_s - mu(T_n - p) * sqrt(face_area) / G.
-
-        Stores this result in self.x which is a ndarray of dimension
-        (3, number of faces). Also updates the ndarray self.is_slipping to
-        True for any face that violated the slip-criterion.
-
-        Requires the following keywords in the data dictionary:
-        'face_pressure': (ndarray) size equal number of faces in the grid.
-                         Only the pressure on the fracture faces are used, and
-                         should be equivalent to the pressure in the
-                        pressure in the corresponding lower dimensional cells.
-        'traction': (ndarray) size (3, number_of_faces). This should be the 
-                    area scaled traction on each face.
-        'rock': a Rock Object with shear stiffness Rock.MU defined.
-
-        Returns:
-            new_slip (bool) returns True if the slip vector was violated for
-                     any faces
-        """
-        assert self._gb.dim == 3, "only support for 3D (yet)"
-
-        frac_faces = self._gb.frac_pairs
-        fi = frac_faces[1]
-        fi_left = frac_faces[0]
-        T_n, T_s, n, t = self.normal_shear_traction(fi)
-
-        assert np.all(T_s > -1e-10)
-        assert np.all(T_n < 0), "Must have a normal force on the fracture"
-
-        # we find the effective normal stress on the fracture face.
-        # Here we need to multiply T_n with -1 as we want the absolute value,
-        # and all the normal tractions should be negative.
-        sigma_n = -T_n - self._data["face_pressure"][fi]
-        #        assert np.all(sigma_n > 0 )
-
-        # new slip are fracture faces slipping in this iteration
-        new_slip = (
-            T_s - self.mu(fi, self.is_slipping[fi]) * sigma_n
-            > 1e-5 * self._data["rock"].MU
-        )
-
-        self.is_slipping[fi] = self.is_slipping[fi] | new_slip
-
-        # calculate the shear stiffness
-        shear_stiffness = np.sqrt(self._gb.face_areas[fi]) / (self._data["rock"].MU)
-
-        # calculate aproximated slip distance
-        excess_shear = np.abs(T_s) - self.mu(fi, self.is_slipping[fi]) * sigma_n
-        slip_d = excess_shear * shear_stiffness * self.gamma() * new_slip
-
-        # We also add the values to the left cells so that when we average the
-        # face values to obtain a cell value, it will equal the face value
-
-        slip_vec = -t * slip_d - n * self.fracture_dilation(slip_d, fi)
-
-        self.d_n[fi] += self.fracture_dilation(slip_d, fi)
-        self.d_n[fi_left] += self.fracture_dilation(slip_d, fi_left)
-
-        assert np.all(self.d_n[fi] > -1e-6)
-
-        self.x[:, fi] += slip_vec
-        self.x[:, fi_left] -= slip_vec
+""" This is a resurrection of a small part of the old fracture_deformation
+model. It contains a single function that should really be placed somewhere els.
```

Comment:
And i'm sure we have three different versions of this hidden away in dark places of the code

Reply:
I will let it live for now, then we can move it later on.